hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/DeletedBlockLogImpl.java

```diff
@@ -386,6 +406,19 @@ public DatanodeDeletedBlockTransactions getTransactions(
             LOG.warn("Container: {} was not found for the transaction: {}.", id, txn);
             txIDs.add(txn.getTxID());
           }
+          if (!iter.hasNext()) {
+            iter.seekToFirst();
+            if (transactions.getBlocksDeleted() >= blockDeletionLimit) {
+              nextProcessTransactionId = -1;
+            }
+          }
+        }
+        if (iter.hasNext() && nextProcessTransactionId != -1) {
+          // Store this value so that next iteration starts from this element instead of beginning
+          nextProcessTransactionId = iter.next().getKey();
+        } else {
+          // No more element in the table, start next iteration from the beginning
+          nextProcessTransactionId = -1L;
```

Comment:
I can understand the idea here after spending some time going over the logic, but this is not easy to follow.